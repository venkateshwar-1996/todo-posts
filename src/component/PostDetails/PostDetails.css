.post{
    display: flex;
    justify-content: center;
    -webkit-touch-callout: none; /* iOS Safari */
    -webkit-user-select: none; /* Safari */
     -khtml-user-select: none; /* Konqueror HTML */
       -moz-user-select: none; /* Firefox */
        -ms-user-select: none; /* Internet Explorer/Edge */
            user-select: none;
}

.postData{
    height: 400px;
}
.userData{
    -webkit-touch-callout: none; /* iOS Safari */
    -webkit-user-select: none; /* Safari */
     -khtml-user-select: none; /* Konqueror HTML */
       -moz-user-select: none; /* Firefox */
        -ms-user-select: none; /* Internet Explorer/Edge */
            user-select: none;
}