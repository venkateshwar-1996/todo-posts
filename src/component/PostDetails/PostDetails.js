import React from "react";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import PropTypes from "prop-types";
import './PostDetails.css';

const Comments = ({ comments }) => {
  const commentsList = (comments || []).map((comment) => {
    return (
      <div key={comment.name}>
        <small className="card-text">Email: {comment.email}</small>
        <p className="card-text">Comment: {comment.body}</p>
        <hr />
      </div>
    );
  });
  return commentsList;
};

const PostDetails = ({ title, body, comments, userName }) => {
  return (
    <div data-testid={'postdetails'}>
      <Row className="userData">
          <h1 className="card-title">{userName}</h1>
      </Row>
      <Row>
        <Col md={8} xs={12} className="card mb-4 bg-dark text-white post">
          <h3 className="card-title">{title}</h3>
          <h5 className="card-text">{body}</h5>
        </Col>
        <Col md={4} xs={12} className="mb-4  postData">
          <div className="comments">
            <Comments comments={comments} />
          </div>
        </Col>
      </Row>
    </div>
  );
};

PostDetails.propTypes = {
  title: PropTypes.string,
  userName: PropTypes.string,
  body: PropTypes.string,
  comments: PropTypes.array,
};
export default PostDetails;
