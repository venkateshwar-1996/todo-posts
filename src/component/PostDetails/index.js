import PostDetails from "./PostDetails";
import withLoadingScreen from './../../HOC/withLoadingScreen';
import { MyInstagramLoader } from './../../helpers/loader';

export default withLoadingScreen(PostDetails, MyInstagramLoader)