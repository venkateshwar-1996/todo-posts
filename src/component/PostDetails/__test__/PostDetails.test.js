import React from "react";
import ReactDOM from "react-dom";
import PostDetails from "../PostDetails";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";

const comments = [
  {
    name: "Venkatesh",
    email: "venkateshwar1996@gmail.com",
    body: "good content",
  },
];

it("Render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <PostDetails
        comments={comments}
        userName={"Venkatesh"}
        body={"content body"}
        id={1}
        title={"React coding"}
      />
    </MemoryRouter>,
    div
  );
});
it("Render PostDetails correctly", () => {
  const { getByTestId } = render(
    <MemoryRouter>
      <PostDetails
        comments={comments}
        userName={"Venkatesh"}
        body={"content body"}
        id={1}
        title={"React coding"}
      />
    </MemoryRouter>
  );
  expect(getByTestId("postdetails")).toMatchSnapshot();
});
it("Render PostDetails correctly", () => {
  const { getByTestId } = render(
    <MemoryRouter>
      <PostDetails
        comments={[]}
        userName={"Venkatesh"}
        body={"content body"}
        id={1}
        title={"React coding"}
      />
    </MemoryRouter>
  );
  expect(getByTestId("postdetails")).toMatchSnapshot();
});
