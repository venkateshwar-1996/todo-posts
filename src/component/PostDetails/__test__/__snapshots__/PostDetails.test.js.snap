// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PostDetails correctly 1`] = `
<div
  class="mt-4"
  data-testid="postdetails"
>
  <div
    class="row"
  >
    <h1
      class="card-title"
    >
      Venkatesh
    </h1>
  </div>
  <div
    class="row"
  >
    <div
      class="card mb-4 bg-dark text-white post col-md-8 col-12"
    >
      <h3
        class="card-title"
      >
        React coding
      </h3>
      <h5
        class="card-text"
      >
        content body
      </h5>
    </div>
    <div
      class="mb-4  postData col-md-4 col-12"
    >
      <div
        class="comments"
      >
        <div>
          <small
            class="card-text"
          >
            Email: 
            venkateshwar1996@gmail.com
          </small>
          <p
            class="card-text"
          >
            Comment: 
            good content
          </p>
          <hr />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render PostDetails correctly 2`] = `
<div
  class="mt-4"
  data-testid="postdetails"
>
  <div
    class="row"
  >
    <h1
      class="card-title"
    >
      Venkatesh
    </h1>
  </div>
  <div
    class="row"
  >
    <div
      class="card mb-4 bg-dark text-white post col-md-8 col-12"
    >
      <h3
        class="card-title"
      >
        React coding
      </h3>
      <h5
        class="card-text"
      >
        content body
      </h5>
    </div>
    <div
      class="mb-4  postData col-md-4 col-12"
    >
      <div
        class="comments"
      />
    </div>
  </div>
</div>
`;
